specs/validator/beacon_node_oapi.yaml

```diff
@@ -0,0 +1,609 @@
+        500:
+          $ref: '#/components/responses/InternalError'
+
+  /validator/duties:
+    get:
+      tags:
+        - MinimalSet
+      summary: "Get validator duties for the requested validators."
+      description: "Requests the beacon node to provide a set of _duties_, which are actions that should be performed by validators. This API call should be polled at every slot, to ensure that any chain reorganisations are catered for, and to ensure that the currently connected beacon node is properly synchronized."
+      parameters:
+      - name: validator_pubkeys
+        in: query
+        required: true
+        description: "An array of hex-encoded BLS public keys"
+        schema:
+          type: array
+          items:
+            $ref: '#/components/schemas/pubkey'
+          minItems: 1
+      responses:
+        200:
+          description: Success response
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/ValidatorDuty'
+        400:
+          $ref: '#/components/responses/InvalidRequest'
+        500:
+          $ref: '#/components/responses/InternalError'
+        503:
+          $ref: '#/components/responses/CurrentlySyncing'
+
+  /validator/block:
+    get:
+      tags:
+        - MinimalSet
+      summary: "Produce a new block, without signature."
+      description: "Requests a beacon node to produce a valid block, which can then be signed by a validator."
+      parameters:
+      - name: slot
+        in: query
+        required: true
+        description: "The slot for which the block should be proposed."
+        schema:
+          type: integer
+          format: uint64
+      - name: randao_reveal
+        in: query
+        required: true
+        description: "The validator's randao reveal value."
+        schema:
+          type: string
+          format: byte
+      responses:
+        200:
+          description: Success response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/BeaconBlock'
+        400:
+          $ref: '#/components/responses/InvalidRequest'
+        500:
+          $ref: '#/components/responses/InternalError'
+        503:
+          $ref: '#/components/responses/CurrentlySyncing'
+    post:
+      tags:
+        - MinimalSet
+      summary: "Publish a signed block."
+      description: "Instructs the beacon node to publish a newly signed beacon block to the beacon network, to be included in the beacon chain."
+      parameters:
+      - name: beacon_block
+        in: query
+        required: true
+        description: "The `BeaconBlock` object, as sent from the beacon node originally, but now with the signature field completed."
+        schema:
+          $ref: '#/components/schemas/BeaconBlock'
+      responses:
+        200:
+          $ref: '#/components/responses/Success'
```

Comment:
Does this response code here mean that beacon node sees the block as valid, has incorporated into local view, and broadcast to network, or does it just mean that it was broadcast?

More specifically, do we expect a BN to validate the incoming block at this point before broadcast or to just assume it is valid and broadcast? In the general case, the BN would likely want to assume it is valid before broadcasting to reduce latency in propagating the block to network.

I suppose the only way a block would be invalid in between this call and response would be if the validator added an invalid sig or if the BN forgot about part of it's block tree between the call and response.  So maybe the validation expected in the response code is just the proposer signature.

The normal reorg scenario is not really an issue because the BN would still see the block as valid in the context of the entire block tree rather than just the head of the chain.

Reply:
After some discussion with @paulhauner - we concluded that in our case, the beacon node will need to validate the incoming block so that it can be incorporated into the state, before it is broadcast. As such, we need to perform validation of the block.

However, we also concluded that it's not the beacon node's responsibility to prevent broadcast of a block if it deems it to be invalid; the validator should know what it's doing.

As such, I have clarified this behaviour in the description, and created a new status code `202` which indicates that the block was published; however it did not pass validity checks (and therefore has not been incorporated into the beacon node's database).